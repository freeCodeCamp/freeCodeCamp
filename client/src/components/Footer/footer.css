/* 11. Site Footer
/* ---------------------------------------------------------- */

.site-footer {
  color: var(--tertiary-color);
  background: var(--tertiary-background);
  line-height: 1.6;
  padding: 40px 15px;
  font-size: 16px;
}

.footer-top {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(30.5em, 1fr));
}

@media (max-width: 500px) {
  .footer-top {
    grid-template-columns: repeat(auto-fit, minmax(19em, 1fr));
  }
}

.footer-top,
.footer-bottom {
  width: min(100%, 1300px);
  margin-inline: auto;
}

.site-footer p {
  margin: 0 0 1.45rem;
  line-height: 30px;
  font-size: 16px;
}

.site-footer a {
  color: var(--tertiary-color);
  text-decoration: none;
  padding: 2px;
}

.site-footer a:hover {
  text-decoration: underline;
}

.site-footer .col-header,
.trending-guides .col-header {
  text-align: center;
  font-weight: 700;
  font-size: 16px;
  padding: 0 15px 15px;
  margin-bottom: 0;
}

.site-footer h2 {
  line-height: 1.1;
}

.trending-guides-articles {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(11em, 1fr));
  grid-column-gap: 3rem;
  grid-row-gap: 0.5rem;
  white-space: pre;
}

@media (max-width: 1400px) {
  .trending-guides-articles {
    grid-template-columns: repeat(auto-fit, minmax(12em, 1fr));
  }
}

@media (max-width: 800px) {
  .trending-guides-articles {
    grid-template-columns: repeat(auto-fit, minmax(13em, 1fr));
  }
}

.footer-desc-col {
  padding-inline: 15px;
  margin-bottom: 30px;
  align-self: start;
}

.footer-desc-col a {
  text-decoration: underline;
  padding: 2px;
}

p.footer-donation {
  font-weight: 700;
  font-size: 18px;
}

p.footer-donation a:hover {
  text-decoration: none;
}

.trending-guides {
  margin: 0 0 3rem;
}

.mobile-app-container {
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 16px;
}

/* This rule to is ensure the li element doesn't take more space than it needs */
.mobile-app-container li {
  display: flex;
  flex: initial;
}

.mobile-app-container img {
  width: 9.5rem;
  height: 3rem;
  object-fit: cover;
}

.footer-bottom .our-nonprofit {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.footer-bottom .our-nonprofit a {
  padding: 5px 10px;
}

@media (min-width: 500px) {
  .mobile-app-container {
    flex-direction: row;
  }
}

@media (min-width: 800px) {
  .footer-bottom .our-nonprofit a {
    padding: 5px;
  }
}

@media (min-width: 1200px) {
  .footer-desc-col {
    flex: 1 0 45%;
  }
  .trending-guides {
    flex: 1 0 58%;
  }
  p.footer-donation {
    margin-top: 40px;
  }
  .footer-bottom .our-nonprofit {
    padding-inline: 10px;
    padding-top: 1.25rem;
    justify-content: space-between;
  }
  .footer-bottom .col-header {
    visibility: hidden;
    padding: 0px;
    margin: 0;
    height: 1px;
  }
  .our-nonprofit {
    border-top: 1px;
    border-top-style: solid;
    border-top-color: var(--quaternary-background);
  }
}
