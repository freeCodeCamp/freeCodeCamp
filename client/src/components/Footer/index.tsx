import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Col, Spacer } from '@freecodecamp/ui';

import appleStoreBadge from '../../assets/images/footer-ads/apple-store-badge.svg';
import googlePlayBadge from '../../assets/images/footer-ads/google-play-badge.svg';
import { Link } from '../helpers';
import './footer.css';

function Footer(): JSX.Element {
  const { t } = useTranslation();

  return (
    <footer className='site-footer'>
      <div className='footer-top'>
        <div className='footer-desc-col'>
          <p>{t('footer.tax-exempt-status')}</p>
          <p>{t('footer.mission-statement')}</p>
          <p>{t('footer.donation-initiatives')}</p>
          <p className='footer-donation'>
            <Trans i18nKey='footer.donate-text'>
              You can
              <Link className='inline' to='/donate'>
                make a tax-deductible donation here
              </Link>
              .
            </Trans>
          </p>
        </div>
        <div className='trending-guides'>
          <h2 id='trending-guides' className='col-header'>
            {t('footer.trending-guides')}
          </h2>
          <ul
            aria-labelledby='trending-guides'
            className='trending-guides-articles'
          >
            <li>
              <Link external={false} to={t('trending:article0link')}>
                {t('trending:article0title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article1link')}>
                {t('trending:article1title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article2link')}>
                {t('trending:article2title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article3link')}>
                {t('trending:article3title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article4link')}>
                {t('trending:article4title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article5link')}>
                {t('trending:article5title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article6link')}>
                {t('trending:article6title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article7link')}>
                {t('trending:article7title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article8link')}>
                {t('trending:article8title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article9link')}>
                {t('trending:article9title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article10link')}>
                {t('trending:article10title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article11link')}>
                {t('trending:article11title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article12link')}>
                {t('trending:article12title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article13link')}>
                {t('trending:article13title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article14link')}>
                {t('trending:article14title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article15link')}>
                {t('trending:article15title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article16link')}>
                {t('trending:article16title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article17link')}>
                {t('trending:article17title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article18link')}>
                {t('trending:article18title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article19link')}>
                {t('trending:article19title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article20link')}>
                {t('trending:article20title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article21link')}>
                {t('trending:article21title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article22link')}>
                {t('trending:article22title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article23link')}>
                {t('trending:article23title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article24link')}>
                {t('trending:article24title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article25link')}>
                {t('trending:article25title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article26link')}>
                {t('trending:article26title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article27link')}>
                {t('trending:article27title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article28link')}>
                {t('trending:article28title')}
              </Link>
            </li>
            <li>
              <Link external={false} to={t('trending:article29link')}>
                {t('trending:article29title')}
              </Link>
            </li>
          </ul>

          <Spacer size='m' />

          <div>
            <h2 id='mobile-app' className='col-header'>
              {t('footer.mobile-app')}
            </h2>
            <Col sm={8} smOffset={2}>
              <ul aria-labelledby='mobile-app' className='mobile-app-container'>
                <li>
                  <Link to='https://apps.apple.com/us/app/freecodecamp/id6446908151?itsct=apps_box_link&itscg=30200'>
                    <img
                      src={appleStoreBadge}
                      lang='en'
                      alt='Download on the App Store'
                    />
                  </Link>
                </li>
                <li>
                  <Link to='https://play.google.com/store/apps/details?id=org.freecodecamp'>
                    <img
                      src={googlePlayBadge}
                      lang='en'
                      alt='Get it on Google Play'
                    />
                  </Link>
                </li>
              </ul>
            </Col>
          </div>
        </div>
      </div>
      <div className='footer-bottom'>
        <h2 className='col-header'>{t('footer.our-nonprofit')}</h2>
        <div className='our-nonprofit'>
          <Link external={false} to={t('links:footer.about-url')}>
            {t('footer.links.about')}
          </Link>
          <Link
            external={false}
            sameTab={false}
            to={'https://www.linkedin.com/school/free-code-camp/people/'}
          >
            {t('footer.links.alumni')}
          </Link>
          <Link external={false} to={'https://github.com/freeCodeCamp/'}>
            {t('footer.links.open-source')}
          </Link>
          <Link
            external={false}
            sameTab={false}
            to={t('links:footer.shop-url')}
          >
            {t('footer.links.shop')}
          </Link>
          <Link external={false} to={t('links:footer.support-url')}>
            {t('footer.links.support')}
          </Link>
          <Link external={false} to={t('links:footer.sponsors-url')}>
            {t('footer.links.sponsors')}
          </Link>
          <Link external={false} to={t('links:footer.honesty-url')}>
            {t('footer.links.honesty')}
          </Link>
          <Link external={false} to={t('links:footer.coc-url')}>
            {t('footer.links.coc')}
          </Link>
          <Link external={false} to={t('links:footer.privacy-url')}>
            {t('footer.links.privacy')}
          </Link>
          <Link external={false} to={t('links:footer.tos-url')}>
            {t('footer.links.tos')}
          </Link>
          <Link external={false} to={t('links:footer.copyright-url')}>
            {t('footer.links.copyright')}
          </Link>
        </div>
      </div>
    </footer>
  );
}

Footer.displayName = 'Footer';
export default Footer;
