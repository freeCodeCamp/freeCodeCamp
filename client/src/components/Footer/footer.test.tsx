import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '.';

describe('<Footer />', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
