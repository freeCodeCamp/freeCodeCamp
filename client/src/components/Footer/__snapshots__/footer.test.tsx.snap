// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<footer
  className="site-footer"
>
  <div
    className="footer-top"
  >
    <div
      className="footer-desc-col"
    >
      <p>
        footer.tax-exempt-status
      </p>
      <p>
        footer.mission-statement
      </p>
      <p>
        footer.donation-initiatives
      </p>
      <p
        className="footer-donation"
      >
        You can
        <a
          className="inline"
          href="/donate"
        >
          make a tax-deductible donation here
        </a>
        .
      </p>
    </div>
    <div
      className="trending-guides"
    >
      <h2
        className="col-header"
        id="trending-guides"
      >
        footer.trending-guides
      </h2>
      <ul
        aria-labelledby="trending-guides"
        className="trending-guides-articles"
      >
        <li>
          <a
            href="trending:article0link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article0title
          </a>
        </li>
        <li>
          <a
            href="trending:article1link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article1title
          </a>
        </li>
        <li>
          <a
            href="trending:article2link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article2title
          </a>
        </li>
        <li>
          <a
            href="trending:article3link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article3title
          </a>
        </li>
        <li>
          <a
            href="trending:article4link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article4title
          </a>
        </li>
        <li>
          <a
            href="trending:article5link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article5title
          </a>
        </li>
        <li>
          <a
            href="trending:article6link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article6title
          </a>
        </li>
        <li>
          <a
            href="trending:article7link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article7title
          </a>
        </li>
        <li>
          <a
            href="trending:article8link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article8title
          </a>
        </li>
        <li>
          <a
            href="trending:article9link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article9title
          </a>
        </li>
        <li>
          <a
            href="trending:article10link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article10title
          </a>
        </li>
        <li>
          <a
            href="trending:article11link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article11title
          </a>
        </li>
        <li>
          <a
            href="trending:article12link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article12title
          </a>
        </li>
        <li>
          <a
            href="trending:article13link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article13title
          </a>
        </li>
        <li>
          <a
            href="trending:article14link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article14title
          </a>
        </li>
        <li>
          <a
            href="trending:article15link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article15title
          </a>
        </li>
        <li>
          <a
            href="trending:article16link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article16title
          </a>
        </li>
        <li>
          <a
            href="trending:article17link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article17title
          </a>
        </li>
        <li>
          <a
            href="trending:article18link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article18title
          </a>
        </li>
        <li>
          <a
            href="trending:article19link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article19title
          </a>
        </li>
        <li>
          <a
            href="trending:article20link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article20title
          </a>
        </li>
        <li>
          <a
            href="trending:article21link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article21title
          </a>
        </li>
        <li>
          <a
            href="trending:article22link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article22title
          </a>
        </li>
        <li>
          <a
            href="trending:article23link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article23title
          </a>
        </li>
        <li>
          <a
            href="trending:article24link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article24title
          </a>
        </li>
        <li>
          <a
            href="trending:article25link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article25title
          </a>
        </li>
        <li>
          <a
            href="trending:article26link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article26title
          </a>
        </li>
        <li>
          <a
            href="trending:article27link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article27title
          </a>
        </li>
        <li>
          <a
            href="trending:article28link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article28title
          </a>
        </li>
        <li>
          <a
            href="trending:article29link"
            rel="noopener noreferrer"
            target="_blank"
          >
            trending:article29title
          </a>
        </li>
      </ul>
      <div
        className="h-[30px]"
      />
      <div>
        <h2
          className="col-header"
          id="mobile-app"
        >
          footer.mobile-app
        </h2>
        <div
          className=" min-h-[1px] px-[15px]  md:w-2/3    md:ml-[16.6%]   "
        >
          <ul
            aria-labelledby="mobile-app"
            className="mobile-app-container"
          >
            <li>
              <a
                href="https://apps.apple.com/us/app/freecodecamp/id6446908151?itsct=apps_box_link&itscg=30200"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Download on the App Store"
                  lang="en"
                  src={{}}
                />
              </a>
            </li>
            <li>
              <a
                href="https://play.google.com/store/apps/details?id=org.freecodecamp"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Get it on Google Play"
                  lang="en"
                  src={{}}
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer-bottom"
  >
    <h2
      className="col-header"
    >
      footer.our-nonprofit
    </h2>
    <div
      className="our-nonprofit"
    >
      <a
        href="links:footer.about-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.about
      </a>
      <a
        href="https://www.linkedin.com/school/free-code-camp/people/"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.alumni
      </a>
      <a
        href="https://github.com/freeCodeCamp/"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.open-source
      </a>
      <a
        href="links:footer.shop-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.shop
      </a>
      <a
        href="links:footer.support-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.support
      </a>
      <a
        href="links:footer.sponsors-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.sponsors
      </a>
      <a
        href="links:footer.honesty-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.honesty
      </a>
      <a
        href="links:footer.coc-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.coc
      </a>
      <a
        href="links:footer.privacy-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.privacy
      </a>
      <a
        href="links:footer.tos-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.tos
      </a>
      <a
        href="links:footer.copyright-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.links.copyright
      </a>
    </div>
  </div>
</footer>
`;
